discovery/roles/configure_ochami/templates/cloud_init/ci-group-slurm_control_node_x86_64.yaml.j2

```diff
@@ -94,13 +94,121 @@
+            mysql -u root < /root/init_slurm_db.sql                 
+
+        - path: /root/omnia_slurm_scripts/02_slurmdbd_setup.sh
+          permissions: '{{ file_mode_755 }}'
+          content: |
+            #!/bin/bash
+            SLURMDBD_CONF="/etc/slurm/slurmdbd.conf"
+            SLURM_USER="{{ user }}"
+            SLURM_GROUP="{{ slurm_group_name }}"
+            # Function to extract value from slurm.conf 
+            get_value_slurm_conf() {
+                local key="$1"
+                local default="$2"
+                local value
+                value=$(grep -iE "^\s*$key\s*=" "$SLURMDBD_CONF" | sed -E 's/^\s*[^=]+=//; s/#.*//; s/\s+$//')
+                echo "${value:-$default}"
+            }
+            chmod {{ file_mode_600 }} /etc/slurm/slurmdbd.conf
+            chown {{ user }}:{{ slurm_group_name }} /etc/slurm/slurmdbd.conf
+            #firewall
+            systemctl enable firewalld
+            systemctl start firewalld
+            DbdPort=$(get_value_slurm_conf "DbdPort" "6819")
+            firewall-cmd --permanent --add-port="$DbdPort"/tcp
+            firewall-cmd --reload
+            systemctl enable slurmdbd
+            systemctl start slurmdbd
+
+        - path: /root/omnia_slurm_scripts/03_ctld_setup.sh
+          permissions: '{{ file_mode_755 }}'
+          content: |
+            #!/bin/bash
+            SLURM_CONF="/etc/slurm/slurm.conf"
+            SLURM_USER="{{ user }}"
+            SLURM_GROUP="{{ slurm_group_name }}"
+            # Function to extract value from slurm.conf 
+            get_value_slurm_conf() {
+                local key="$1"
+                local default="$2"
+                local value
+                value=$(grep -iE "^\s*$key\s*=" "$SLURM_CONF" | sed -E 's/^\s*[^=]+=//; s/#.*//; s/\s+$//')
+                echo "${value:-$default}"
+            }
+            #dir StateSaveLocation
+            StateSaveLocation=$(get_value_slurm_conf "StateSaveLocation" "/var/spool")
+            mkdir -pv $StateSaveLocation
+            chown -v "$SLURM_USER:$SLURM_GROUP" $StateSaveLocation
+            chmod -v 0744 $StateSaveLocation
+            #file SlurmctldPidFile
+            SlurmctldPidFile=$(get_value_slurm_conf "SlurmctldPidFile" "/var/run/slurmctld.pid")
+            mkdir -pv $(dirname "$SlurmctldPidFile")
+            touch $SlurmctldPidFile
+            chown -v "$SLURM_USER:$SLURM_GROUP" "$SlurmctldPidFile"
+            chmod -v 0755 $SlurmctldPidFile
+            #file SlurmctldLogFile
+            SlurmctldLogFile=$(get_value_slurm_conf "SlurmctldLogFile" "/var/log/slurmctld.log")
+            mkdir -pv $(dirname "$SlurmctldLogFile")
+            touch $SlurmctldLogFile
+            chown -v "$SLURM_USER:$SLURM_GROUP" "$SlurmctldLogFile"
+            chmod -v 0755 $SlurmctldLogFile
+            #firewall
+            systemctl enable firewalld
+            systemctl start firewalld
+            SlurmctldPort=$(get_value_slurm_conf "SlurmctldPort" "6817")
```

Comment:
can we take all the ports from a var file

Reply:
no, this is run time fetched so that when conf is changed this scripts will do the needful